### DISCUSSION TITLE
Next-Auth Prisma Adapter Client TypeError

### DISCUSSION THREAD

**Discussion by masnwilliams** [prisma/prisma#11664] [Q&A] [answered]
Hello! I created an instance of Prisma in `lib/prisma.ts` inside my next.js project to use across the entire project. However, when using the prisma instance for Next-Auth, I get a type error. If I create a new instance for Next-Auth then it works but then I find myself maxing out connections when developing and would assume that it would open a new connection for every user that uses next-auth to signin/sign out/etc in production as well.

![image](https://user-images.githubusercontent.com/43387599/152698268-c519c06f-4ba4-4863-97e2-173f61ea9072.png)

This is the code inside `lib/prisma.ts` for my prisma instance:
```
import { PrismaClient } from '@prisma/client'

let prisma

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient()
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient()
  }

  prisma = global.prisma
}

export default prisma
```

Thank you.

**Comment by masnwilliams** [comment, marked as answer]
Update: I solved the issue.

I was just importing prisma incorrectly for Next-Auth.

Error Import:
`import { prisma } from '@prisma/client'`

Corrected Import:
`import prisma from '@lib/prisma'`

Thank you. This can be closed now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
